test/js/ui/marker.test.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+var test = require('tap').test;
+var jsdom = require('jsdom');
+
+var extend = require('../../../js/util/util').extend;
+var Map = require('../../../js/ui/map');
+var Marker = require('../../../js/ui/marker');
+var Popup = require('../../../js/ui/popup');
+
+function prepareDOM(html) {
+    html = html || '<!doctype html><html><body><div id="map"></div></body></html>';
+    if (typeof document !== 'undefined') {
+        return;
+    }
+    global.document = jsdom.jsdom(html);
+    global.window = global.document.defaultView;
+    global.navigator = {
+        userAgent: 'JSDOM'
+    };
+    global.HTMLElement = global.window.HTMLElement;
+
+}
```

Comment:
Can you refactor this so that we don't need to use global variables? You should be able to do everything you need with access to the `window` object within test-scope.